src/modules/launcher/Wox/PublicAPIInstance.cs

```diff
@@ -69,6 +70,9 @@ public void CheckForNewUpdate()
 
         public void SaveAppAllSettings()
         {
+            StorageHelper _storageHelper = new StorageHelper();
```

Comment:
The encapsulation is a bit weird to me.   This is coupling the two implementations to a single file.  If that is the intent, than perhaps the StorageHelper should be injected into the BinaryStorageWriter and JsonStorageWriter.  If we want to keep them seperate, then this should probably done at the end of a save operation.

Reply:
I agree it's a little hacky. The reason I didn't do this at the end of a Save operation is because we would be unnecessarily re-writing the version multiple times. It's more of a one time operation. I could have a different class implement this but it would have to get the version number and filePath again. I tried to distinguish the two code paths based on whether the class has a constructor with or without arguments. Do you have any suggestions on better ways to go about it?

Reply:
Based on your comment above, if we want to have multiple version numbers for each storage file, then I could call the `close` function in the `Save` method of each storage object. Just to clarify that we're both on the same page, then we would have a txt file corresponding to each json/cache storage file?

Reply:
Made the version specific to each storage file.